tests/models/test_token_classification_with_seq2seq_encoder_and_crf.py

```diff
@@ -0,0 +1,421 @@
+        monkeypatch=monkeypatch,
+        model_type="bert",
+        batch_size=inputs["input_ids"].shape[0],
+        seq_len=inputs["input_ids"].shape[1],
+        num_classes=int(torch.max(targets) + 1),
+        seq2seq_encoder=seq2seq_dict,
+    )
+    return model
+
+
+@pytest.fixture
+def model_with_ce(monkeypatch, batch):
+    inputs, targets = batch
+    seq2seq_dict = {
+        "type": "linear",
+        "out_features": 10,
+    }
+    model = get_model(
+        monkeypatch=monkeypatch,
+        model_type="bert",
+        batch_size=inputs["input_ids"].shape[0],
+        seq_len=inputs["input_ids"].shape[1],
+        num_classes=int(torch.max(targets) + 1),
+        seq2seq_encoder=seq2seq_dict,
+        use_crf=False,
+    )
+    return model
+
+
+@pytest.mark.parametrize(
+    "model_type", list(HF_MODEL_TYPE_TO_CLASSIFIER_DROPOUT_ATTRIBUTE) + ["unknown"]
+)
+def test_config_model_classifier_dropout(monkeypatch, model_type):
+    if model_type in HF_MODEL_TYPE_TO_CLASSIFIER_DROPOUT_ATTRIBUTE:
+        model = get_model(
+            monkeypatch=monkeypatch,
+            model_type=model_type,
+            batch_size=4,
+            seq_len=10,
+            num_classes=5,
+        )
+        assert model is not None
+        assert isinstance(model, TokenClassificationModelWithSeq2SeqEncoderAndCrf)
+    else:
+        with pytest.raises(ValueError):
+            model = get_model(
+                monkeypatch=monkeypatch,
+                model_type=model_type,
+                batch_size=4,
+                seq_len=10,
+                num_classes=5,
+            )
+
+
+@pytest.mark.parametrize("seq2seq_enc_type", list(RNN_TYPE2CLASS))
+def test_seq2seq_classification_head(monkeypatch, seq2seq_enc_type):
+    seq2seq_dict = {
+        "type": seq2seq_enc_type,
+        "hidden_size": 10,
+    }
+    model = get_model(
+        monkeypatch,
+        model_type="bert",
+        batch_size=4,
+        seq_len=10,
+        num_classes=5,
+        freeze_base_model=True,
+        seq2seq_encoder=seq2seq_dict,
+    )
+
+    assert model is not None
+    assert isinstance(model, TokenClassificationModelWithSeq2SeqEncoderAndCrf)
+    assert isinstance(model.seq2seq_encoder.rnn, RNN_TYPE2CLASS[seq2seq_enc_type])
+
+
+def test_freeze_base_model(monkeypatch, batch):
+    inputs, target = batch
+    # set seed to make the classifier deterministic
+    model = get_model(
+        monkeypatch,
+        model_type="bert",
+        batch_size=4,
+        seq_len=10,
+        num_classes=5,
+        add_dummy_linear=True,
+        freeze_base_model=True,
+    )
+    base_model_params = list(model.model.parameters())
+    # the dummy linear from the mock base model has 2 parameters
+    assert len(base_model_params) == 2
+    for param in base_model_params:
+        assert not param.requires_grad
+
+    classification_head_params = list(model.parameters())[2:]
+    for param in classification_head_params:
+        assert param.requires_grad
```

Comment:
having `classification_head_params = list(model.parameters())[2:]` is quite prone to errors, because when you change the order of parameters in the model, this will fail. I would propose sth like:
```suggestion
    base_model_params = dict(model.model.named_parameters(prefix="model"))
    assert len(base_model_params) > 0
    for param in base_model_params.values():
        assert not param.requires_grad

    head_params = {
        name: param for name, param in model.named_parameters() if name not in base_model_params
    }
    assert len(head_params) > 0
    for param in head_params.values():
        assert param.requires_grad
```

Reply:
i actually thought of this, but should the base model always be the first layers? having anything before the base model should fail the assertion?